cpp/src/arrow/type.cc

```diff
@@ -1544,7 +1544,7 @@ Result<std::shared_ptr<Schema>> Schema::AddField(
 Result<std::shared_ptr<Schema>> Schema::SetField(
     int i, const std::shared_ptr<Field>& field) const {
   if (i < 0 || i > this->num_fields()) {
-    return Status::Invalid("Invalid column index to set field.");
+    return Status::Invalid("Invalid column index to add field.");
```

Comment:
Since this change was part of a revert commit, the structure of the code for `SetField` ie exactly the same as the code for `AddField` above it, and since it doesn't make much sense to me either, I'm guessing this was a derp from git's merge/rebase algorithm. a8dd172